richtextfx/src/main/java/org/fxmisc/richtext/model/TextChange.java

```diff
@@ -10,12 +10,16 @@
         INSERTION,
         /** Indicates that the change will delete something but not insert anything */
         DELETION,
-        /** Indicates that the change will remove something and insert something as its replacement */
-        REPLACEMENT
+        /** Indicates that the change will remove something and insert something as its replacement.
+         * For {@link PlainTextChange}s, the replacement will only be content.
+         * For {@link RichTextChange}s, the replacement may only be the style, only the content, or a mix of both.
+         * Style changes may change the style of the paragraph's content or a non-empty {@link Paragraph}
+         * ({@code p.length() > 0}) */
+        REPLACEMENT,
+        /** Indicates that the paragraph style of an empty {@link Paragraph} ({@code p.length() == 0}) was changed */
+        RESTYLED_PARAGRAPH
```

Comment:
Thinking about this now, I think this should be named "RESTYLED_EMPTY_PARAGRAPH" to be clearer and prevent confusion (e.g. a user thinks it occurs when a non-empty paragraph is restyled